### PR TITLE
O11Y: Added support for custom tags

### PR DESCRIPTION
**Pull request by zasweq**
This PR adds support for custom tags in Observability for Metrics, Tracing, and Logging. Custom tags are defined in the configuration and attached to emitted Logs, Traces, and Metrics. Currently there is no way to test this E2E, so the best we can do is verify the config passed into the Tracing and Metrics exporter has the correct custom tags present. Also, since this added a new field to the configuration, we decided it would be better if we got rid of the implementation detail of representation the configuration with a proto file, which I switched to an internal struct.

This PR does the following:
* Deletes support for the previous way you set custom tags, by setting Environment Variables with "GRPC_OBSERVABILITY_" as a prefix
* Rewrote the testing injection logic for Metrics and Tracing, previously pass in exporter, now global function you overwrite
* Adds new custom tags field in configuration and attaches them to any emitted Metrics, Traces, or Logs.
* Switch proto representation of configuration to an internal go struct

RELEASE NOTES:
* O11y: Add support for custom tags

### CODE REVIEW

File: gcp/observability/config.go [hunk 1]
```diff
@@ -41,6 +39,69 @@ const (
 
 var logFilterPatternRegexp = regexp.MustCompile(logFilterPatternRegexpStr)
 
+// LogFilter represents a method logging configuration.
+type LogFilter struct {
```
Comment (dfawley) [inline gcp/observability/config.go:43]: This type should not be exported, right?  Same below.

File: gcp/observability/config.go [hunk 2]
```diff
@@ -41,6 +39,69 @@ const (
...
+	// of the message is greater than the defined limit, content pass the limit
+	// will be truncated. The default value is 0.
+	MessageBytes int32 `json:"message_bytes,omitempty"`
+}
+
+// ObvConfig is configuration for observability behaviors. By default, no
+// configuration is required for tracing/metrics/logging to function. This
+// config captures the most common knobs for gRPC users. It's always possible to
+// override with explicit config in code.
+type ObvConfig struct {
```
Comment (dfawley) [inline gcp/observability/config.go:74]: Just `config`?  Or maybe `jsonConfig`?

File: gcp/observability/opencensus.go [hunk 3]
```diff
@@ -38,26 +37,59 @@ var (
 	defaultMetricsReportingInterval = time.Second * 30
 )
 
+func convertTagsToMonitoringLabels(tags map[string]string) *stackdriver.Labels {
```
Comment (dfawley) [inline gcp/observability/opencensus.go:40]: Drop the `convert`; it's cleaner?

File: gcp/observability/config.go (hunk 2 above)
Comment (zasweq) [reply gcp/observability/config.go]: Chose config.

File: gcp/observability/opencensus.go (hunk 3 above)
Comment (zasweq) [reply gcp/observability/opencensus.go]: Done.

**Review by zasweq — COMMENTED** [review]
Thanks for the comments :D!

File: gcp/observability/config.go (hunk 1 above)
Comment (zasweq) [reply gcp/observability/config.go]: Done.

**Review by dfawley — APPROVED** [review]

**Pull request merged**